resources/scenarios/ln_init.py

```
@@ -188,7 +199,9 @@ def connect_ln(self, pair):
         # so their channel ids are deterministic
         ch_by_block = {}
         for ch in self.channels:
-            # TODO: if "id" not in ch ...
+            if "id" not in ch or "block" not in ch["id"]:
+                self.log.info(f"LN Channel {ch} not found")
+                continue
```

I think this is another place where we shouldn't just silently fail. User wants graph, user gets graph or we abort.

In fact, I'm wondering why you had to make so many changes to `ln_init.py` at all. Everything lnd-specific *should* have been already abstracted out. What would fail on your branch if this file was totally restored? Maybe if you have ln_init optimizations they can go in a separate PR?

---

I'll revert the safeguards I added and they can go in a future PR.